conjure-java-core/src/main/java/com/palantir/conjure/java/types/AliasGenerator.java

```diff
@@ -446,15 +449,17 @@ private static CodeBlock primitiveSafeEquality(
     }
 
     private static CodeBlock primitiveSafeToString(TypeName aliasTypeName) {
-        if (aliasTypeName.isPrimitive()) {
+        if (PrimitiveHelpers.isPrimitive(aliasTypeName)) {
             return CodeBlocks.statement("return $T.valueOf(value)", String.class);
         }
         return CodeBlocks.statement("return value.toString()");
     }
 
     private static CodeBlock primitiveSafeHashCode(TypeName aliasTypeName) {
-        if (aliasTypeName.isPrimitive()) {
-            return CodeBlocks.statement("return $T.hashCode(value)", aliasTypeName.box());
+        if (PrimitiveHelpers.isPrimitive(aliasTypeName)) {
+            return CodeBlocks.statement(
+                    "return $T.hashCode(value)",
+                    PrimitiveHelpers.box(aliasTypeName).withoutAnnotations());
```

Comment:
before you could add safety to external longs, you were never going to end up with an aliasTypeName that is primitive for all intents and purposes (e.g. java.lang.Long) but is annotated. now, that situation is possible so we need to strip the annotation here.